src/Tizen.Multimedia.Vision/MediaVision/InferenceFaceRecognier.cs

```diff
@@ -0,0 +1,175 @@
+        /// Registers face data to internal database.
+        /// </summary>
+        /// <param name="source">The face data to register.</param>
+        /// <param name="label">The name of face data.</param>
+        /// <exception cref="ObjectDisposedException">The InferenceFaceRecognizer already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> or <paramref name="label"/> is null.</exception>
+        /// <exception cref="InvalidOperationException">In case of any invalid operations.</exception>
+        /// <since_tizen> 10 </since_tizen>
+        public void RegisterFace(MediaVisionSource source, string label)
+        {
+            ValidateNotDisposed();
+
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+            if (label == null)
+            {
+                throw new ArgumentNullException(nameof(label));
+            }
+
+            InteropFace.Register(_handle, source.Handle, label).Validate("Failed to register face data");
+        }
+
+        /// <summary>
+        /// Unregisters face data from internal database.
+        /// </summary>
+        /// <param name="label">The name of face data.</param>
+        /// <exception cref="ObjectDisposedException">The InferenceFaceRecognizer already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="label"/> is null.</exception>
+        /// <exception cref="InvalidOperationException">In case of any invalid operations.</exception>
+        /// <since_tizen> 10 </since_tizen>
+        public void UnregisterFace(string label)
+        {
+            ValidateNotDisposed();
+
+            if (label == null)
+            {
+                throw new ArgumentNullException(nameof(label));
+            }
+
+            InteropFace.Unregister(_handle, label).Validate("Failed to register face data");
+        }
+
+        /// <summary>
+        /// Recognizes input face from internal face datas and returns its label.
+        /// </summary>
+        /// <param name="source">The face data to recognize.</param>
+        /// <exception cref="ObjectDisposedException">The InferenceFaceRecognizer already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        /// <exception cref="InvalidOperationException">In case of any invalid operations.</exception>
+        /// <since_tizen> 10 </since_tizen>
+        public string Recognize(MediaVisionSource source)
+        {
+            ValidateNotDisposed();
+
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            InteropFace.Inference(_handle, source.Handle).Validate("Failed to recognize face");
+
+            InteropFace.GetLabel(_handle, out string label).Validate("Failed to get label");
```

Comment:
Inference and GetLatel methods could return some exception so how about adding try-catch statement like below?
try
{
InteropFace.Inference(_handle, source.Handle).Validate("Failed to recognize face");
InteropFace.GetLabel(_handle, out string label).Validate("Failed to get label");
}
catch (Exception e)
{
Log.Error(MediaVisionLog.Tag, e.ToString());
throw e;
}

return label;

Reply:
I could be done with Validate() method.